apps/main/src/lend/components/DetailInfoRate.tsx

```diff
@@ -16,11 +18,22 @@ const DetailInfoRate = ({
   isBorrow: boolean
   futureRates?: FutureRates | undefined | null
 }) => {
+  // uses onchain data if available
+  const { data: onChainData } = useMarketOnChainRates({ chainId: rChainId, marketId: rOwmId })
   const ratesResp = useStore((state) => state.markets.ratesMapper[rChainId]?.[rOwmId])
 
   const { rates, error } = ratesResp ?? {}
   const futureRate = isBorrow ? futureRates?.borrowApy : futureRates?.lendApr
 
+  const rate = useMemo(() => {
```

Comment:
Sorry for the confusion, what I meant here is that we should just call `useSupplyTotalApr` and let the hook decide where the data comes from. Basically, deciding where data comes from does not belong in an UI component but it's business logic